src/main/java/io/github/hotlava03/chatutils/mixin/JoinServerMixin.java

```diff
@@ -1,16 +1,28 @@
 package io.github.hotlava03.chatutils.mixin;
 
 import io.github.hotlava03.chatutils.events.JoinServerEvent;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.hud.ChatHud;
 import net.minecraft.client.gui.screen.ConnectScreen;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
+import java.util.List;
+
 @Mixin(ConnectScreen.class)
 public class JoinServerMixin {
+    @Mixin(ChatHud.class)
+    public interface MessageHistoryAccessor {
+        @Accessor
+        List<String> getMessageHistory();
+    }
+
```

Comment:
Why is this here and not in its own mixin (use this mixin from the listener itself), JoinServerEvent has nothing to do with message history.